runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcDirectGetWorkStream.java

```diff
@@ -240,25 +234,46 @@ public void sendHealthCheck() {
   @Override
   protected void onResponse(StreamingGetWorkResponseChunk chunk) {
     getWorkThrottleTimer.stop();
-    WorkItemBuffer workItemBuffer =
-        workItemBuffers.computeIfAbsent(chunk.getStreamId(), unused -> new WorkItemBuffer());
+    GetWorkItemBuffer workItemBuffer =
+        workItemBuffers.computeIfAbsent(chunk.getStreamId(), unused -> new GetWorkItemBuffer());
     workItemBuffer.append(chunk);
 
     // The entire WorkItem has been received, it is ready to be processed.
     if (chunk.getRemainingBytesForWorkItem() == 0) {
```

Comment:
it seems a little nicer if the assembler did this since it already gets the chunk.

Can append be changed to return the work item if it is completed? the size could be communicated in that Constructed/AssembledWorkItem object

Reply:
done